ably/ably.go

```diff
@@ -0,0 +1,24 @@
+package ably
+
+func min(i, j int) int {
+	if i < j {
+		return i
+	}
+	return j
+}
+
+func max(i, j int) int {
+	if i > j {
+		return i
+	}
+	return j
+}
+
+func nonil(err ...error) error {
```

Comment:
Question for you guys: Does Golang have any tool to detect unused private method (dead code)?

It would be nice to have that in our CI too :smile:.